ramalama/cli.py

```diff
@@ -775,7 +774,7 @@ def push_cli(args):
                 raise e
         try:
             # attempt to push as a container image
-            m = OCI(tgt, config.get('engine', container_manager()))
+            m = ModelFactory(tgt, engine=config.get('engine', container_manager())).create_oci()
```

Comment:
It's probably not for this PR, but it would be better if we resolved all environment variables, configuration files, CLI args, once early in execution of ramalama. I started that here at one point, but it drifted:

load_and_merge_config

we tend to do things like, use the "engine" value or container_manager() over and over again in this codebase which isn't great.

Reply:
I agree and good to know. I'll keep that in mind and will try to refactor occurrences like this.